### ISSUE TITLE
Inconsistent values between cursor hover display and stack watch display when select a specific frame in the stack view

### ISSUE THREAD

**Issue by tan-wei** [pkulchenko/ZeroBraneStudio#1001]
For example, `k` is `nil` when cursor hover, however in the stack watch window, `k` is `packet`, which is the correct one.
![image](https://user-images.githubusercontent.com/28227509/61139111-47ceea00-a4fb-11e9-8556-222774e85756.png)

Version: 1.80, Linux-x64

**Comment by pkulchenko** [comment]
@tan-wei, I don't see where the debugger is suspended, but the tooltip is *always* executed in the context of the current application position and the stack evaluation has more information available (as it looks at the various levels of variables that the evaluation doesn't have access to). I suspect that in the location where the debugging is suspended, there is no `k` variable defined/available, so you see `nil` as the value returned.

**Comment by tan-wei** [comment]
Yes, `k` is defined in the lowest stack. So the cursor hover display value is **always** the value in current stack, we can't select a stack frame to change it. I'm used to `MSVC`, it behaves like I expect.
So I think it's not a bug instead of an imagine feature by myself, which is not always necessary. Hence the issue is closed.

**Comment by pkulchenko** [comment]
> So the cursor hover display value is always the value in current stack, we can't select a stack frame to change it.

@tan-wei, correct. See #717 for the previous discussion on the same topic (I'll mark this ticket as a duplicate). Some of the comments in that ticket no longer apply, as it *is* now possible to execute a watch in a particule stack frame context.

Are you suggesting that the selected stack frame should be used to set the context? What if you select one and then execute Step or Run command, which will change the stack content? Should this "selection" work only for the current context? I'm not sure how it works in MSVC.

Basically, if you just want to execute watch in the context of the currently selected stack frame, it should be fairly straightforward to implement, with the assumption that the stack will be reset after the next command, so you'll need to select the frame again. If no frame is selected, then the current one is used (as it is now).

**Comment by tan-wei** [comment]
For `MSVC`, the selected stack frame is used to set the context Unfortunately, I don't know about the implement details about the feature.

The current status is fine for me, the feature about cursor hover display is not a necessary one.
